packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SubjectGroupViewer/components/SGVGridCell/SGVGridCell.js

```diff
@@ -0,0 +1,128 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import { draggable } from '@plugins/drawingTools/components'
+
+const DraggableImage = draggable('image')
+const DraggableRect = draggable('rect')
+
+function SGVGridCell (props) {
+  const {
+    image,
+    index,
+    
+    dragMove,
+    
+    cellWidth,
+    cellHeight,
+    cellStyle,
+    gridRows,
+    gridColumns,
+    
+    panX,
+    panY,
+    zoom,
+  } = props
+    
+  const row = Math.floor(index / gridColumns)
+  const col = index % gridColumns
+
+  const cellXOffset = col * cellWidth
+  const cellYOffset = row * cellHeight
+
+  if (
+    !image || !image.src || !image.naturalHeight || !image.naturalWidth  // Don't render an image if there's no image to render. Of course.
+    || row < 0 || row >= gridRows || col < 0 || col >= gridColumns  // Don't render anything beyond the specified grid.
+    || !cellWidth || !cellHeight || !gridColumns || !gridRows
+  ) return null
+
+  // TODO: what if there are fewer images than cells in the grid?
+
+  const fitRatio = Math.max(
+    image.naturalWidth / cellWidth,
+    image.naturalHeight / cellHeight,
+  )
+
+  const imageHeight = image.naturalHeight / fitRatio
+  const imageWidth = image.naturalWidth / fitRatio
+
+  // image.x and image.y determine the default 'padding' for an image inside
+  // its cell, and is applied before the zoom & translation/pan transforms.
+  // Note: this COULD be consolidated into the transform calculations, but why
+  // complicate things?
+  const imageX = (cellWidth - imageWidth) / 2
+  const imageY = (cellHeight - imageHeight) / 2
+
+  const clipPathID = `subjectGroupViewer-clipPath-${index}`
+
+  return (
+    <g
+      transform={`translate(${cellXOffset}, ${cellYOffset})`}
+    >
+      <clipPath id={clipPathID}>
+        <rect width={cellWidth} height={cellHeight} />
+      </clipPath>
+      <DraggableRect
+        fill={cellStyle.fill}
+        width={cellWidth}
+        height={cellHeight}
+        dragMove={dragMove}
+      />
+      <g clipPath={`url(#${clipPathID})`}>
+        <DraggableImage
+          dragMove={dragMove}
+          height={imageHeight}
+          width={imageWidth}
+          xlinkHref={image.src}
```

Comment:
Depending on the size of the original images, should we pass them through the thumbnail service?

Reply:
I've noted this as a separate issue - https://github.com/zooniverse/front-end-monorepo/issues/1584

The ability to pass images to the thumbnail service is a good idea, but I think should be configurable, and default to false at the start, since some researchers may actually be providing hi-res Subject images so they can ask users to zoom in on the details.

Reply:
With our rough limit of ~600k per image, that gives a download of ~5.4MB per subject for a cluster of 9 images. 😨